src/Otis.Users/S3Client.cs

```diff
@@ -0,0 +1,30 @@
+using Amazon;
+using Amazon.S3;
+using Amazon.S3.Model;
+
+namespace Otis.Users;
+
+public interface IS3Client
+{
+    Task UploadFileToBucket(string bucketName, string keyName, string content);
+}
+
+public sealed class S3Client : IS3Client
```

Comment:
It would be nice to put every pre-registration in the same file. I don't know how hard it is, but if too complicated ignore this comment then. Easier for manual analysis to download the entire file than have multiple files to scrap for info

Reply:
It's not about being complicated, but more about potential issues caused by concurrent access to a single file.
That's the reason why I chose to have one file per request: to avoid concurrency issues

I'll investigate if S3 offers a solution for that and come back later today

Reply:
@Higino we also have an issue regarding the file format because 

> JSON is a non-concatenative protocol (the concatenation of two JSON objects does not produce a valid JSON object).
> -- [Wikipedia](https://en.wikipedia.org/wiki/JSON_streaming)

Did some investigation and came up with options

# OPTION 1: Newline-delimited JSON [option in JSON Streaming](https://en.wikipedia.org/wiki/JSON_streaming) 
Stores each JSON object in a single line of a file.

✅ Adding a new entity appends a new line at the end of the file (requires some locking mechanism to avoid concurrent writes)
✅ `jq` tool supports this method
✅ counting the lines of the file yields the number of records 
❌ JSON fields CAN NOT contain new lines (which is not a problem in our case
⚠️  Requires a `lock` mechanism to avoid concurrent access when writing to the file

# OPTION 2: Store data in a JSON Array
Stores each JSON object in JSON Array
✅ Produces a valid JSON file
⚠️ Requires reading the whole file to memory, append the data and overwrite the content again
⚠️ Requires a `lock` mechanism to avoid concurrent access when writing to the file
💀 This might cause data loss if the lock is poorly implemented

# OPTION 3: One file per record (current implementation)
✅ Avoids locking
✅ Easy to concat to JSON Lines format and use `jq` or custom tool to aggregate records 

# My recommendation

I'd recommend to stick with the current implementation because it's a simpler solution and we can easily crunch data when we want to process it.

If you'd like to have aggregated data from day 1, I'd suggest using a database for that purpose.

Thoughts?

Reply:
Lets keep with the current implementation. I agree we can easily craft a solution that puts the json of each file into a table and compute some analytics over it.